src/cloudai/systems/slurm/strategy/slurm_command_gen_strategy.py

```diff
@@ -175,20 +176,48 @@ def _write_sbatch_script(self, args: Dict[str, Any], env_vars_str: str, srun_com
             f"#SBATCH -N {args['num_nodes']}",
         ]
 
+        self._append_sbatch_directives(batch_script_content, args, output_path)
+
+        batch_script_content.append(env_vars_str)
+        batch_script_content.append(srun_command)
+
+        script_path = os.path.join(output_path, "cloudai_sbatch_script.sh")
+        with open(script_path, "w") as script_file:
+            script_file.write("\n".join(batch_script_content))
+
+        return f"sbatch {script_path}"
+
+    def _append_sbatch_directives(
+        self, batch_script_content: List[str], args: Dict[str, Any], output_path: str
+    ) -> None:
+        """
+        Append SBATCH directives to the batch script content.
+
+        Args:
+            batch_script_content (List[str]): The list of script lines to append to.
+            args (Dict[str, Any]): Arguments including job settings.
+            output_path (str): Output directory for script and logs.
+        """
         if "output" not in args:
             batch_script_content.append(f"#SBATCH --output={os.path.join(output_path, 'stdout.txt')}")
         if "error" not in args:
             batch_script_content.append(f"#SBATCH --error={os.path.join(output_path, 'stderr.txt')}")
-        if args["partition"]:
+
+        # safely access parition
```

Comment:
Not sure if it is needed, `_parse_slurm_args` raises if partition is not set, it doesn't make sense writing sbatch script without it. Not important, but the comment is not needed here.

Reply:
If we don't specify the partition, it will go to default.  In CW, I let my account access to define that for me. So it works btw. That line was already there.  So I don't see why there was a check  exists if it didn't make sense to write one without partition in the first place. 

https://github.com/NVIDIA/cloudai/blob/main/src/cloudai/systems/slurm/strategy/slurm_command_gen_strategy.py#L182

This is just an additional check. I don't see it doing any harm.

Reply:
I believe partition is required for Cloud AI internals, so this check was introduced. 

Anyway, what I mean is that using safe `agrs.get(...)` doesn't really add any safety compared to `args[...]` because it is anyway verified before.

Not super important, but double-check could be confusing for readers of the code.

Reply:
IMO getter method handles the key error and returns None instead if throwing KeyError in the latter. Anyway not inclined either way. Reverted back.